presto-main/src/test/java/com/facebook/presto/execution/TestSetPathTask.java

```diff
@@ -0,0 +1,127 @@
+import com.facebook.presto.metadata.SessionPropertyManager;
+import com.facebook.presto.metadata.TablePropertyManager;
+import com.facebook.presto.security.AccessControl;
+import com.facebook.presto.security.AllowAllAccessControl;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.facebook.presto.sql.tree.Identifier;
+import com.facebook.presto.sql.tree.PathElement;
+import com.facebook.presto.sql.tree.PathSpecification;
+import com.facebook.presto.sql.tree.SetPath;
+import com.facebook.presto.transaction.TransactionManager;
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+
+import static com.facebook.presto.SessionTestUtils.TEST_SESSION;
+import static com.facebook.presto.transaction.TransactionManager.createTestTransactionManager;
+import static io.airlift.concurrent.MoreFutures.getFutureValue;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static org.testng.Assert.assertEquals;
+
+public class TestSetPathTask
+{
+    private final TransactionManager transactionManager;
+    private final AccessControl accessControl;
+    private final MetadataManager metadata;
+
+    private final ExecutorService executor = newCachedThreadPool(daemonThreadsNamed("stage-executor-%s"));
+
+    public TestSetPathTask()
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        transactionManager = createTestTransactionManager(catalogManager);
+        accessControl = new AllowAllAccessControl();
+
+        metadata = new MetadataManager(
+                new FeaturesConfig(),
+                new TypeRegistry(),
+                new BlockEncodingManager(new TypeRegistry()),
+                new SessionPropertyManager(),
+                new SchemaPropertyManager(),
+                new TablePropertyManager(),
+                transactionManager);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testSetPath()
+    {
+        PathSpecification pathSpecification = new PathSpecification(
+                Optional.empty(), ImmutableList.of(
+                        new PathElement(Optional.empty(), new Identifier("foo"))));
+
+        QueryStateMachine stateMachine = createQueryStateMachine("SET PATH foo");
+        executeSetPathTask(pathSpecification, stateMachine);
+
+        assertEquals(stateMachine.getSetPath(), "foo");
+    }
+
+    @Test (expectedExceptions = PrestoException.class, expectedExceptionsMessageRegExp = "Catalog does not exist: .*")
```

Comment:
Nit: no space between the annotation and the opening `(`

(this will probably be caught by Checkstyle)